### PR TITLE
[MRG+1] travis deployment with tags

### PR DESCRIPTION
**Pull request by eLRuLL**
now we need to create a tag, for travis deployment to pypi.

### CODE REVIEW

**Comment by pablohoffman** [comment]
@eliasdorneles you have experience with deploys, can you review and merge this one?

It would be nice to have tests running on py3 as well.

File: .travis.yml [hunk 1]
```diff
@@ -6,3 +6,14 @@ env:
...
 script: tox
+deploy:
+  provider: pypi
+  distributions: "sdist bdist_wheel"
+  user: scrapinghub
+  password:
+    secure: bG1ycgBJrVfgBsHAwaWScL2V1x52/aZy2znE2kFzFfYq95HS51mu1GO2PKwpATJyMCs/xUIKGhiBtoo8LzNmNU2kPgarQFwzpLkx2ninOm+3uAwuIfGGWsonBW/h854n+cNwppJOlfEOxvMVerVVpdV4EFeJfEkVYascwOBbnqo=
+  on:
+    tags: true
+    all_branches: true
```
Comment (eliasdorneles) [inline .travis.yml]: it's best to remove this line, so that releases are made only for master branch.

File: .travis.yml [hunk 2]
```diff
@@ -6,3 +6,14 @@ env:
 install:
   - pip install -U tox
 script: tox
+deploy:
+  provider: pypi
+  distributions: "sdist bdist_wheel"
+  user: scrapinghub
+  password:
+    secure: bG1ycgBJrVfgBsHAwaWScL2V1x52/aZy2znE2kFzFfYq95HS51mu1GO2PKwpATJyMCs/xUIKGhiBtoo8LzNmNU2kPgarQFwzpLkx2ninOm+3uAwuIfGGWsonBW/h854n+cNwppJOlfEOxvMVerVVpdV4EFeJfEkVYascwOBbnqo=
```
Comment (eliasdorneles) [inline .travis.yml]: was this password generated for this repo with `travis` cmdline tool?

File: .travis.yml (hunk 1 above)
Comment (dangra) [inline .travis.yml]: +1 that line was included as a workaround for travis-ci/travis-ci/issues/1675.

**Comment by eliasdorneles** [comment]
@eLRuLL I recommend using bumpversion to do the releases.

You can create a `.bumpversion.cfg` config file with this:

```
[bumpversion]
current_version = 1.0.2
commit = True
tag = True
tag_name = v{new_version}

[bumpversion:file:setup.py]

[bumpversion:file:scrapy-crawlera/__init__.py]
```

and then you can use the bumpversion utility to create the releases (update version number and tag the commits).

**Comment by pablohoffman** [comment]
@eliasdorneles shouldn't we document about `bumpversion` in `internaldoc`? /cc @dangra

**Comment by dangra** [comment]
Yes, bumpversion+travis become our standard tool to publish new packages to
pypi

2015-08-04 14:23 GMT-03:00 Pablo Hoffman <notifications@github.com>:

**Comment by eLRuLL** [comment]
could we get `current_version` dynamically on bumpversion with from `setup.py` file?

**Comment by eLRuLL** [comment]
why do we need `[bumpversion:file:scrapy-crawlera/__init__.py]`?, and that file doesn't exist.

**Comment by dangra** [comment]
> why do we need [bumpversion:file:scrapy-crawlera/__init__.py]?, and that file doesn't exist.

you usually want to know what version is installed by importing the python module and looking at __version__ attribute. It is also useful to customize Proxy-User-agent header (if any ;))

**Comment by dangra** [comment]
bumpversion helps you keep value in setup.py and in python module in sync

**Comment by eliasdorneles** [comment]
@eLRuLL oh, you're right, I dind't check, sorry! Anyway, I second @dangra's comments. :)
Do you think it would be useful to have a `__version__` attribute in `scrapy_crawlera.py`?

@pablohoffman +1, I'll send a PR to internaldoc soon.

**Comment by eLRuLL** [comment]
so we are duplicating the `__version__` attribute? it already exists on `setup.py`

**Comment by dangra** [comment]
> so we are duplicating the __version__ attribute? it already exists on setup.py

right. but you can't import setup.py to extract `__version__` and importing `scrapy_crawlera.py` from setup.py to extract __version__ won't work neither.

**Comment by eliasdorneles** [comment]
@eLRuLL yeah, that kind of duplication (static version number in installation script and inside the module) is normal and sort of the reason bumpversion exists.
When you run `bumpversion minor`, it will increase the version number for a minor release (like, if it was 0.1.0 it will be 0.2.0) in all configured files (including its own config), create a commit and tag it.

Btw, it seems the changes in `.travis.yml` are gone (can't see them in the diff) -- did something went wrong in a rebase or something?

**Comment by eLRuLL** [comment]
@eliasdorneles sorry travis is added now. About the travis key, I just copied the one from (open source deployment)[https://github.com/scrapinghub/internaldoc/blob/master/open-sourcing-projects.md#deploying-to-pypi] , when can I get the one from this project?

**Comment by eliasdorneles** [comment]
@eLRuLL you need to install the travis command line tool (it's a Ruby gem, so `gem install travis`), and then, inside your local repo, run: `travis encrypt --add deploy.password` and follow the on-screen instructions.
I think @dangra can give you the password for pypi deployment -- not sure which user should be used for this.

**Comment by dangra** [comment]
@eLRuLL uses the following user and password:
```
deploy:
  user: scrapinghub
  password:
    secure: rU/nBv26TZbZhO22zDeSWtQjmglSeG+ujWqfYUiLuAl4+JcNnQSFtqK14S4CZOLCjl8lfIJjZV073+X2zhy0AUVQrNlbVFRCwnAL9KsDG3s9ixGNhNdZB82R0+VdViI39M+Fhty8X1g0XhPmAoDz7Ol9G5Voyu5gCINQHFCaAxs=
```

**Comment by eLRuLL** [comment]
thanks @dangra, @eliasdorneles is there something else needed?

**Comment by eliasdorneles** [comment]
@eLRuLL that's it, LGTM!

**Pull request merged**